src/query/api/v1/middleware/rewrite.go

```diff
@@ -163,21 +181,31 @@ func rewriteRangeDuration(
 	updatedURL.RawQuery = urlQueryValues.Encode()
 	r.URL = updatedURL
 
-	// Update query param in the request body, if present
+	// Update query and lookback params in the request body, if present and needed.
 	if r.Form.Get(queryParam) != "" {
-		r.Form.Set(queryParam, updatedQuery)
+		if updateQuery {
+			r.Form.Set(queryParam, updatedQuery)
+		}
+		if updateLookback {
+			r.Form.Set(lookbackParam, resolutionBasedLookback.String())
+		}
 	}
 
-	logger.Debug("rewrote range duration value within query",
+	logger.Debug("rewrote duration values in request",
+		zap.Bool("updateQuery", updateQuery),
 		zap.String("originalQuery", query),
-		zap.String("updatedQuery", updatedQuery))
+		zap.String("updatedQuery", updatedQuery),
+		zap.Bool("updateLookback", updateLookback),
+		zap.Bool("isLookbackSet", isLookackSet),
+		zap.Duration("originalLookback", lookback),
+		zap.Duration("updatedLookback", resolutionBasedLookback))
 
 	return nil
 }
 
-func extractParams(r *http.Request, instant bool) (string, time.Time, time.Time, error) {
+func extractParams(r *http.Request, instant bool) (string, time.Time, time.Time, time.Duration, bool, error) {
```

Comment:
nit: It would be great to have at least a comment with all the names for all those multiple return values . Comment would be visible in the hint box IDE displays when hovering on the function name so there would be no need to look into function body itself.

Reply:
https://github.com/m3db/m3/pull/3882/commits/bcf48423c7fa0f20d0f80c8f4ea44532c5c4d8e8#diff-f284760ffa1aa45c70f82bdf3d8982cc7328962c64802875e8cf7c5dd9c61784R235-R241